File: abc/abc221/a.java
// https://atcoder.jp/contests/abc221/submissions/41532663

import java.util.Scanner;

class Main {
  public static void main(String[] args) {
    Scanner stdIn = new Scanner(System.in);

    int a = stdIn.nextInt();
    int b = stdIn.nextInt();
    int c = a - b;

    int ans = 1;
    while(c > 0) {
      ans *= 32;
      c--;
    }

    System.out.println(ans);
  }
}

File: abc/abc221/a.cs
// https://atcoder.jp/contests/abc221/submissions/46509003

using System;

public class A
{
  static void Main()
  {
    string[] t = Console.ReadLine().Split(" ");
    int a = int.Parse(t[0]);
    int b = int.Parse(t[1]);

    int ans = 1;
    for(int i = 0; i < a - b; i++)
    {
      ans *= 32;
    }

    Console.Write(ans);
  }
}

